XenModel/Actions/Network/ChangeNetworkingAction.cs

```diff
@@ -235,5 +246,75 @@ private string GetIPInRange(string range_start, PIF existing_pif)
             string[] bits = range_start.Split('.');
             return string.Format("{0}.{1}.{2}.{3}", bits[0], bits[1], bits[2], int.Parse(bits[3]) + i);
         }
+
+        /// <summary>
+        /// Disable clustering on the host (if the network is used by clustering), before changing the management interface; 
+        /// Before disabling clustering we also unplug all the GFS2 SRs
+        /// </summary>
+        private void DisableClustering(PIF pif, out List<PBD> gfs2Pbds)
+        {
+            gfs2Pbds = new List<PBD>();
+            var isUsedByClustering = Connection.Cache.Clusters.Any(cluster => cluster.network.opaque_ref == pif.network.opaque_ref);
+            if (!isUsedByClustering)
+                return;
+
+            var host = Connection.Resolve(pif.host);
+            if (host == null)
+                return;
+
+            var clusterHost = Connection.Cache.Cluster_hosts.FirstOrDefault(c => c.host.opaque_ref == host.opaque_ref);
+            if (clusterHost == null) 
+                return;
+
+            // unplug the GFS2 SRs, saving the list of the PBDs unplugged, to plug back later
+            foreach (var pbd in Connection.ResolveAll(host.PBDs).Where(pbd => pbd.currently_attached))
+            {
+                var sr = Connection.Resolve(pbd.SR);
+                if (sr != null && sr.type.ToLowerInvariant() == "gfs2")

```

Comment:
I think "gfs2" should be defined as a constant globally. It's also used in ClusteringEditPage.cs

Reply:
Good point! Actually, we should use SR.GetSRType(), like `sr.GetSRType(true) == SR.SRTypes.gfs2`

Reply:
even better, thanks

Reply:
@GaborApatiNagy please see the new commit